webnn/concat.https.any.js

```diff
@@ -0,0 +1,48 @@
+// META: title=test WebNN API concat operation
+// META: global=window,dedicatedworker
+// META: script=./resources/utils.js
+// META: timeout=long
+
+'use strict';
+
+// https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-concat
+
+const concatTests = () => {
```

Comment:
Hi @huningxin, thanks for your reviewing and suggestions.
Since there're different implementations (L#16 to L#29) to set `targetTests` for WebNN operations, for example the implementation for `gemmTests` likes:
```js
  for (const test of tests) {
    const gemmOptions = {};
    const options = test.options;
    if (options !== undefined) {
      if (options.c !== undefined) {
        gemmOptions.c = {data: inputsData[options.c.data]};
        if (options.c.shape !== undefined) {
          gemmOptions.c['shape'] = options.c.shape;
        }
      }
      if (options.alpha !== undefined) {
        gemmOptions.alpha = inputsData[options.alpha.data];
      }
      if (options.beta !== undefined) {
        gemmOptions.beta = inputsData[options.beta.data];
      }
      if (options.aTranspose !== undefined) {
        gemmOptions.aTranspose = options.aTranspose;
      }
      if (options.bTranspose !== undefined) {
        gemmOptions.bTranspose = options.bTranspose;
      }
    }
    targetTests.push(
      {
        name: test.name,
        operandType: test.type,
        inputA: {shape: test.inputs.a.shape, data: inputsData[test.inputs.a.data]},
        inputB: {shape: test.inputs.b.shape, data: inputsData[test.inputs.b.data]},
        expected: {shape: test.expected.shape, data: {outputOperand: expectedData[test.expected.data]}},
        options: gemmOptions
      }
    );
  }
```
and implementation for `clampTests` likes:
```js
  for (const test of tests) {
    const expectedDataSource = test.expected.data;
    const inputDataSource = test.input.data;
    targetTests.push(
      {
        name: test.name,
        operandType: test.type,
        input: {shape: test.input.shape, data: inputsData[inputDataSource]},
        expected: {shape: test.expected.shape, data: {outputOperand: expectedData[expectedDataSource]}},
        options: test.options
      }
    );
  }
```

 so I implemented each `<operation-name>Tests` function for each operation, thanks.

Reply:
@huningxin I just submitted tests for clamp (#37067) / gemm (#37068) / matmul (#37069) / reshape (#37070) / relu (#37071) / slice (#37072) operations, PTAL, thanks.

Reply:
This function looks like mainly setting up the test object for `targetTests`. It seems to me that all information and configuration of each test object come from the JOSN file. I am wondering: could we just define all these info in the JSON file and load it into `resources`. I suppose we don't need different JS implementation to setup test object for different ops.

Did I miss anything?

Reply:
@huningxin Do you suggest add actual test data for "data"? For example https://github.com/web-platform-tests/wpt/blob/32290b240b50ef914a03af71f9136b805d177b10/webnn/resources/test_data/concat.json#L8

my purpose of using the index string key of input / expected data is to reuse test data, thanks.

Reply:
@huningxin I've updated this PR for more reuse code, please take another look. And there's a [reusable tests demo](https://github.com/BruceDai/wpt/pull/442/files) on my own repo for your quickly review, thanks.